app/ui/lib/Combobox.tsx

```diff
@@ -93,17 +93,18 @@ export const Combobox = ({
         disabled={disabled || isLoading}
       >
         {label && (
-          // TODO: FieldLabel needs a real ID
           <div className="mb-2">
             <FieldLabel
-              id="FieldLabel"
+              id={`${id}-label`}
```

Comment:
If this doesn't end up getting passed to some other component as `aria-labelled-by` then it's probably not needed at all, which is nice. I was poking around the HTML to see what's going on and it seems like the `Label` component from Headless handles this all for you, taking its text contents and making sure they're set as the label for the corresponding input. See the 

<img width="684" alt="image" src="https://github.com/user-attachments/assets/807b9562-c844-4fd5-9349-aeefbff72c09">

The only obstacle to removing the prop entirely is that it's required on `FieldLabel` in order to try to help us not forget to do these things in cases where Headless isn't doing it for us. That's probably still desirable, so I'm thinking that rather than making `id` optional on `FieldLabel`, the best thing is to leave this as-is but add a comment saying it's not actually used and Headless is taking care of it.